pkg/cmd/extension/manager.go

```diff
@@ -468,21 +468,28 @@ func (m *Manager) upgradeExtensions(exts []Extension, force bool) error {
 			fmt.Fprintf(m.io.Out, "%s\n", err)
 			continue
 		}
-		fmt.Fprintf(m.io.Out, "upgrade complete\n")
+		if dryRun {
+			fmt.Fprintf(m.io.Out, "would have upgraded from %s to %s\n", f.currentVersion, f.latestVersion)
+		} else {
+			fmt.Fprintf(m.io.Out, "upgrade complete\n")
+		}
 	}
 	if failed {
 		return errors.New("some extensions failed to upgrade")
 	}
 	return nil
 }
 
-func (m *Manager) upgradeExtension(ext Extension, force bool) error {
+func (m *Manager) upgradeExtension(ext Extension, force, dryRun bool) error {
 	if ext.isLocal {
 		return localExtensionUpgradeError
 	}
 	if !ext.UpdateAvailable() {
 		return upToDateError
 	}
+	if dryRun {
+		return nil
```

Comment:
I feel like this exits too soon in the dry-run mode. There is heavy logic ahead that I can imagine it would be good to exercise even in dry-run mode. A dry-run should ideally do as much as a normal run would, sans any mutations (which get printed instead of executed).

Reply:
@mislav I updated this PR to now exercise more of the logic during dry-run mode. I opted against printing mutations for two reasons. First, in extensions commands we already don't display what is happening behind the scenes and this decision felt like keeping consistency. Second, when using the `--all` flag the output did not look clean with multiple output lines per extension that would be upgraded.